src/messaging/application/process-reminder-email.js

```diff
@@ -0,0 +1,271 @@
+  db,
+  logger
+) => {
+  const notClaimedNineMonths = await getApplicationsWithoutClaimAfterNineMonths(
+    requestedDate,
+    maxBatchSize,
+    db,
+    logger
+  )
+  const notClaimedSixMonths = await getApplicationsWithoutClaimAfterSixMonths(
+    requestedDate,
+    maxBatchSize,
+    db,
+    logger
+  )
+  const notClaimedThreeMonths =
+    await getApplicationsWithoutClaimAfterThreeMonths(
+      requestedDate,
+      maxBatchSize,
+      db,
+      logger
+    )
+
+  const remindersNotClaimed = [
+    ...notClaimedNineMonths,
+    ...notClaimedSixMonths,
+    ...notClaimedThreeMonths
+  ]
+    .slice(0, maxBatchSize)
+    .map(unwrapDatabaseQueryDataValues)
+    .map(removeOrgEmailIfSameAddressAsEmail)
+    .map(promoteToNextReminderIfNoRemindersAndWithinOneMonth)
+
+  return [...remindersNotClaimed]
+}
+
+const getApplicationsWithoutClaimAfterNineMonths = async (
+  requestedDate,
+  maxBatchSize,
+  db,
+  logger
+) => {
+  const { nineMonths } = reminderTypes.notClaimed
+  const NINE_MONTHS = 9
+
+  const nineMonthReminderWindowStart = new Date(requestedDate)
+  nineMonthReminderWindowStart.setUTCMonth(
+    nineMonthReminderWindowStart.getMonth() - NINE_MONTHS
+  )
+
+  return getRemindersToSend(
+    nineMonths,
+    nineMonthReminderWindowStart,
+    undefined,
+    [],
+    maxBatchSize,
+    db,
+    logger
+  )
+}
+
+const getApplicationsWithoutClaimAfterSixMonths = async (
+  requestedDate,
+  maxBatchSize,
+  db,
+  logger
+) => {
+  const { sixMonths, nineMonths } = reminderTypes.notClaimed
+  const SIX_MONTHS = 6
+  const NINE_MONTHS = 9
+
+  const sixMonthReminderWindowStart = new Date(requestedDate)
+  sixMonthReminderWindowStart.setUTCMonth(
+    sixMonthReminderWindowStart.getMonth() - SIX_MONTHS
+  )
+  const sixMonthReminderWindowEnd = new Date(requestedDate)
+  sixMonthReminderWindowEnd.setUTCMonth(
+    sixMonthReminderWindowEnd.getMonth() - NINE_MONTHS
+  )
+
+  return getRemindersToSend(
+    sixMonths,
+    sixMonthReminderWindowStart,
+    sixMonthReminderWindowEnd,
+    [nineMonths],
+    maxBatchSize,
+    db,
+    logger
+  )
+}
+
+const getApplicationsWithoutClaimAfterThreeMonths = async (
+  requestedDate,
+  maxBatchSize,
+  db,
+  logger
+) => {
+  const { threeMonths, sixMonths, nineMonths } = reminderTypes.notClaimed
+  const THREE_MONTHS = 3
+  const SIX_MONTHS = 6
+
+  const threeMonthReminderWindowStart = new Date(requestedDate)
+  threeMonthReminderWindowStart.setUTCMonth(
+    threeMonthReminderWindowStart.getMonth() - THREE_MONTHS
+  )
+  const threeMonthReminderWindowEnd = new Date(requestedDate)
+  threeMonthReminderWindowEnd.setUTCMonth(
+    threeMonthReminderWindowEnd.getMonth() - SIX_MONTHS
+  )
+
+  return getRemindersToSend(
+    threeMonths,
+    threeMonthReminderWindowStart,
+    threeMonthReminderWindowEnd,
+    [sixMonths, nineMonths],
+    maxBatchSize,
+    db,
+    logger
+  )
+}
+
+const unwrapDatabaseQueryDataValues = (reminder) => {
+  return { ...reminder.dataValues }
```

Comment:
This looks like a hangover from sequelize, don't think it's needed for MongoDB document